### ISSUE TITLE
Rule for rest spread missuse

### ISSUE THREAD

**Issue by graingert** [eslint/eslint#8118]
**Please describe what the rule should do:**
Prevent rest/spread overcomplication 
**What category of rule is this? (place an "X" next to just one item)**

[x] Enforces code style
[ ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
// bad
const foo = {
    ...{a, value: x + 3},
    ...stuff,
};

const bar = [
   ...[a, b, c],
   ...otherStuff,
]

foo(...[a, b, c])

function foo(a, b, ...[c, d]) {
}

// good
const foo = {
    a,
    value: x + 3,
    ...stuff,
};

const bar = [
   a,
   b,
   c,
   ...otherStuff,
]

foo(a, b, c)

function foo(a, b, c, d) {
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**
I wanted to include this rule in https://github.com/bryanrsmith/eslint-plugin-prefer-object-spread see https://github.com/bryanrsmith/eslint-plugin-prefer-object-spread/issues/45#issuecomment-281185216
however @bryanrsmith thought it would be better as a core rule.

> Good idea. I think it would be good to create a new rule for that rather than extend this one. I'd expect ESLint core to start accepting rules around object rest/spread soon, now that the proposal is pretty far along and is starting to be implemented by vendors, so I suggest opening an issue in the main ESLint repo.

**Comment by mysticatea** [comment]
Thank you for this issue.

I have an implement in my plugin: https://github.com/mysticatea/eslint-plugin/blob/master/docs/rules/no-useless-rest-spread.md
If the core-team is accepted this, I can send the implement to core.

**Comment by graingert** [comment]
@mysticatea does your rule do object rest/spread?

**Comment by mysticatea** [comment]
Ah, no, sorry.

**Comment by graingert** [comment]
@mysticatea the timeline for this in core is when it hits Stage 4, correct?

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.

**Comment by graingert** [comment]
@not-an-aardvark fixed by https://github.com/eslint/eslint/pull/9955
